internal/v1/handler_blueprints.go

```diff
@@ -59,11 +66,29 @@ func (h *Handlers) CreateBlueprint(ctx echo.Context) error {
 		return err
 	}
 
+	if !regexp.MustCompile(blueprintNameRegex).MatchString(blueprintRequest.Name) {
+		return ctx.JSON(http.StatusUnprocessableEntity, HTTPErrorList{
+			Errors: []HTTPError{{
+				Title:  "Invalid blueprint name",
+				Detail: "The blueprint name must start and end with an alphanumeric character and may contain alphanumeric characters, underscores, hyphens, and spaces.",
+			}},
+		})
+	}
+
 	id := uuid.New()
 	versionId := uuid.New()
 	ctx.Logger().Infof("Inserting blueprint: %s (%s), for orgID: %s and account: %s", blueprintRequest.Name, id, userID.OrgID(), userID.AccountNumber())
 	err = h.server.db.InsertBlueprint(ctx.Request().Context(), id, versionId, userID.OrgID(), userID.AccountNumber(), blueprintRequest.Name, blueprintRequest.Description, body)
 	if err != nil {
+		var e *pgconn.PgError
+		if errors.As(err, &e) && e.Code == pgerrcode.UniqueViolation {
+			return ctx.JSON(http.StatusUnprocessableEntity, HTTPErrorList{
+				Errors: []HTTPError{{
+					Title:  "Name not unique",
+					Detail: "A blueprint with the same name already exists.",
+				}},
+			})
+		}
 		ctx.Logger().Errorf("Error inserting id into db: %s", err.Error())
```

Comment:
Move the log statement to the beginning of the block so it is logged on unique violation too.